ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -240,32 +288,77 @@ def status(self) -> dict:
             pcs_cluster_status = PcsClusterStatus()
             pcs_cluster_status.load()
-            return pcs_cluster_status.get_health_status()
+            return pcs_cluster_status.get_cluster_health()
         except Exception as e:
             Log.error(f"Failed to get status of the cluster. Error: {e}")
-            return {"status": const.STATUSES.FAILED, "output": "Retry Suggested.", "error" : str(e)}
+            return {"status": const.STATUSES.FAILED.value, "output": "Retry Suggested.", "error" : str(e)}
+
 
     @controller_error_handler
     def standby(self, sync=False, timeout=30) -> dict:
         """
         Put cluster in standby mode.
 
+        Args:
+            sync (bool, optional): if sync is True then stop will check the status for timeout seconds.
+            timeout (int, optional): timeout(in seconds) can be specified for sync=True otherwise ignored.
+
         Returns:
             ([dict]): Return dictionary. {"status": "", "output":"", "error":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        if not self._is_pcs_cluster_running():
+            return {"status": const.STATUSES.FAILED.value, "output": "Node should be online.",
+                    "error": "Cluster is not running on current node."}
+
+        node_group: list = self._get_node_group()
+        for node_subgroup in node_group:
+            for node_id in node_subgroup:
+                self._controllers[const.NODE_CONTROLLER].standby(node_id)
+
+        if sync:
+            timeout = timeout - const.BASE_WAIT_TIME*const.PCS_NODE_GROUP_SIZE*len(node_group)
+            in_expected_state = self._verify_expected_cluster_status(const.CLUSTER_STATUS.STANDBY, timeout)
+            if in_expected_state:
+                return {"status": const.STATUSES.SUCCEEDED.value, "output": "Cluster is on standby.", "error": ""}
+            else:
+                return {"status": const.STATUSES.FAILED.value, "output": "Retry suggested.", "error": "Operation timed out."}
+
+        return {"status": const.STATUSES.IN_PROGRESS.value, "output": "Operation is in progress.", "error": ""}
 
     @controller_error_handler
-    def active(self, sync=False, timeout=30) -> dict:
+    def active(self, sync=False, timeout=90) -> dict:
         """
         Activate all node.
 
+        Args:
+            sync (bool, optional): if sync is True then stop will check the status for timeout seconds.
+            timeout (int, optional): timeout(in seconds) can be specified for sync=True otherwise ignored.
+
         Returns:
             ([dict]): Return dictionary. {"status": "", "output":"", "error":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        status_result = OperationResult(self.status())
+        if status_result.get_output() != const.CLUSTER_STATUS.STANDBY.value:
```

Comment:
if some nodes are in standby state and some are in another state (ex offline etc) how will that be handled ?
And Is it a valid scenario ?

Reply:
Cluster is in standby only when all the nodes are in standby. For any other scenario - cluster start can be used.